chain/client/src/client_actor.rs

```diff
@@ -1484,6 +1509,67 @@ impl ClientActor {
     }
 }
 
+struct StatePartsActor {
```

Comment:
I think we might want to change the name to be more generic to include other jobs we want to separate from `ClientActor`, such as splitting states and block catchup. I am not super good at making names, but maybe something like `BackgroundClientActor` could work (probably not the best name, but hopefully the message I want to convey gets through). The idea is that we can put all job that might block `ClientActor` to there.

Similarly, the messages between should also be an enum. It can be something like

`
pub enum BackgroundJobRequest {
    ApplyStatePartsRequest(ApplyStatePartsRequest),
    SplitStateRequest(SplitStateRequest),
    BlockCatchupRequest(BlockCatchupRequest),
}
`

Reply:
I am not sure we will use same actor for those jobs

Reply:
as an aside, we can just implement Handler trait for different messages without need for enum

Reply:
I am not sure about block catchup, but I think it makes sense to put split state into the same actor. This way, we can have fewer actors in the code to maintain. But maybe it can be done at another PR.

Reply:
Good point. I agree that it probably makes sense to use the same actor for splitting state and applying state parts

Reply:
Yep, I think any renaming should be done when this actor will acquire new functions